File: backend/src/main/java/com/server/backend/question/Question.java
package com.server.backend.question;

import lombok.Data;
import lombok.NoArgsConstructor;
import org.jetbrains.annotations.NotNull;

import javax.persistence.*;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotEmpty;
import java.util.List;
import java.util.Set;

@Entity
@Data
@NoArgsConstructor
public class Question {

    @Id
    @GeneratedValue
    private Integer id;

    @NotNull
    @Column(unique = true)
    private String question;

    @NotNull
    private String correctOption;

    @NotNull
    @NotEmpty
    @ElementCollection
    private List<String> wrongOptions;

    @NotNull
    private String difficulty;

}

File: node-backend/src/Types/Question.ts
export type Question = {
  id: number;
  question: string;
  correct_option: string;
  difficulty: string;
};

export type QuestionWrongOption = {
  id: number;
  question_id: number;
  wrong_option: string;
};

// DTOS
export type QuestionDTO = {
  id: number;
  question: string;
  correct_option: string;
  wrong_options: [string, string, string];
  difficulty: string;
};
